docs/DafnyRef/Grammar.md

```diff
@@ -562,6 +562,52 @@ but the names of the parameters are optional. This is used in algebraic
 datatype definitions.
 
 ### 2.6.5. Quantifier Domains {#sec-quantifier-domains}
+
+Several Dafny constructs bind one or more variables to a range of possible values.
+For example, the quantifier `forall x: nat | x <= 5 :: x * x <= 25` has the meaning
+"for all integers x between 0 and 5, the square of x is at most 25".
+Similarly, the set comprehension `set x: nat | x <= 5 :: f(x)` can be read as
+"the set containing the result of applying f to x, for each integer x between 0 and 5".
+The common syntax that specifies the bound variables and what values they take on
+is known as the *quantifier domain*; in the previous examples this is `x: nat | x <= 5`, 
+which binds the variable `x` to the values `0`, `1`, `2`, `3`, `4`, and `5`.
+
+Here are some more examples.
+
+- `x: byte` (where `type byte = x | 0 <= x < 256`)
+- `x: nat | x <= 5`
+- `x <- integerSet`
+- `x: nat <- integerSet`
+- `x: nat <- integerSet | x % 2 == 0`
+- `x: nat, y: nat | x < 2 && y < 2`
```

Comment:
```suggestion
- `x: nat, y: nat | x < 2 && y < 2`
- `x: nat  | x < 2, y: nat | y < x`
```
It would be good to have a simpler example just before the next one, because the next one uses vertical bars for a different meaning.

Reply:
Yeah I like this, more of a gradual onramp of complexity :)